toolchain/sem_ir/builtin_function_kind.def

```diff
@@ -19,6 +19,9 @@
 
 CARBON_SEM_IR_BUILTIN_FUNCTION_KIND(None)
 
+// A noop function definition; calls should be elided.
```

Comment:
Tweak to (I think?) better match your PR description:

```suggestion
// A noop function definition; calls should be elided when lowering.
```

I like the strategy of not fully eliding these in the SemIR so we're being *explicit* about the no-op.